Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/refine_mesh_at_isolevel.h

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2021-2023 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sebastien Loriot
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_REFINE_MESH_AT_ISOLEVEL_H
+#define CGAL_POLYGON_MESH_PROCESSING_REFINE_MESH_AT_ISOLEVEL_H
+
+#include <CGAL/license/Polygon_mesh_processing/miscellaneous.h>
+
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+/*!
+ * \ingroup PkgPolygonMeshProcessingRef
+ *
+ * refines `pm` by adding new vertices on edges having their incident vertices associated with
+ * values respectively larger and smaller than `isovalue` in `value_map`.
+ * The placement of new vertices on edges will be done by linear interpolation
+ * using the aforementioned values.
+ * New vertices will be associated `isovalue` in `value_map` when created.
+ * Additionally, new edges will be added by connecting new vertices created sharing
+ * a common incident face. Note that in case more than two new vertices are added
+ * on a face boundary, no edges will be created in that face.
+ *
+ * @tparam PolygonMesh a model of the concepts `EdgeListGraph` and `FaceListGraph`
+ * @tparam ValueMap a model of the concept `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+ *                  as key type and with its value type being the type of the coordinates of points associated with vertices
+ *                  in the vertex map provided to the `vertex_point_map()` named parameter.
+ * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters" for `pm`
+ *
+ * @param pm the polygon mesh to be refined.
+ * @param value_map the property map containing a value at each vertex for a given function defined over the mesh.
+ * @param isovalue the value used to refine
+ * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamNBegin{edge_is_constrained_map}
+ *     \cgalParamDescription{an output property map associating `true` to all new edges connecting vertices on the isolevel,
```

Comment:
```suggestion
 *     \cgalParamDescription{an output property map associating `true` to all edges connecting vertices on the isolevel,
```

Reply:
I suggest to remove new, as there may also be input edges that are on the isolevel.